[BUG] Image build pipeline fails

**Describe the bug**
smoke test fail in the build image deploy pipeline:

ghcr.io/pengine-ai/pengine-brave-search@sha256:e452cd4f0e7fdf1d74052327442e59584cc37ec33dc6d3c999425283c142104c: Pulling from pengine-ai/pengine-brave-search
d17f077ada11: Pulling fs layer
2d5b9bf56ef8: Pulling fs layer
e153bc545a79: Pulling fs layer
9a52281c988a: Pulling fs layer
5e2b8519e3fe: Pulling fs layer
b7ee405468ef: Pulling fs layer
48c75bbcc5c9: Pulling fs layer
9a52281c988a: Waiting
5e2b8519e3fe: Waiting
b7ee405468ef: Waiting
48c75bbcc5c9: Waiting
e153bc545a79: Verifying Checksum
e153bc545a79: Download complete
d17f077ada11: Download complete
d17f077ada11: Pull complete
2d5b9bf56ef8: Verifying Checksum
2d5b9bf56ef8: Download complete
5e2b8519e3fe: Verifying Checksum
5e2b8519e3fe: Download complete
9a52281c988a: Download complete
b7ee405468ef: Verifying Checksum
b7ee405468ef: Download complete
48c75bbcc5c9: Verifying Checksum
48c75bbcc5c9: Download complete
2d5b9bf56ef8: Pull complete
e153bc545a79: Pull complete
9a52281c988a: Pull complete
5e2b8519e3fe: Pull complete
b7ee405468ef: Pull complete
48c75bbcc5c9: Pull complete
Digest: sha256:e452cd4f0e7fdf1d74052327442e59584cc37ec33dc6d3c999425283c142104c
Status: Downloaded newer image for ghcr.io/pengine-ai/pengine-brave-search@sha256:e452cd4f0e7fdf1d74052327442e59584cc37ec33dc6d3c999425283c142104c
ghcr.io/pengine-ai/pengine-brave-search@sha256:e452cd4f0e7fdf1d74052327442e59584cc37ec33dc6d3c999425283c142104c
Error: --brave-api-key is required. You can get one at [https://brave.com/search/api/.](https://brave.com/search/api/)
Invalid configuration
Error: Process completed with exit code 1.

**To Reproduce**
Steps to reproduce the behavior:

1. Go to '...'
2. Click on '....'
3. Scroll down to '....'
4. See error

**Expected behavior**
All test should work and images should be deployed but only when the mcp server docker images change in the mcp-tools.json list.

--- a/.github/scripts/tools-publish/smoke-test-mcp.sh
+++ b/.github/scripts/tools-publish/smoke-test-mcp.sh
@@ -16,8 +16,14 @@ if [[ "${TOOL_SLUG:-}" == "file-manager" ]]; then
   extra=(/tmp)
 fi
 
+# Brave Search MCP refuses to start with an empty key; init handshake does not call the API.
+docker_run_env=()
+if [[ "${TOOL_SLUG:-}" == "brave-search" ]]; then
+  docker_run_env=(-e "BRAVE_API_KEY=${SMOKE_BRAVE_API_KEY:-smoke-ci-placeholder}")
+fi
+
 RESP=$(echo '{"jsonrpc":"2.0","id":1,"method":"initialize","params":{"protocolVersion":"2024-11-05","capabilities":{},"clientInfo":{"name":"smoke","version":"0.0.1"}}}' \
-  | timeout 15 docker run --rm -i --network=none "$IMAGE_WITH_DIGEST" "${extra[@]}" \
+  | timeout 15 docker run --rm -i --network=none "${docker_run_env[@]}" "$IMAGE_WITH_DIGEST" "${extra[@]}" \
   | head -1)
 echo "$RESP" | jq -e '.result.serverInfo' > /dev/null \
   || { echo "::error::MCP init failed: $RESP"; exit 1; }

